lib/fluent/plugin/out_papertrail.rb

```diff
@@ -62,16 +63,22 @@ def send_to_papertrail(packet)
       # recreate the socket if it's nil -- see below
       @socket ||= create_socket(@papertrail_host, @papertrail_port)
 
+      papertrail_addr = "#{@papertrail_host}:#{@papertrail_port}"
+
       if @socket.nil?
-        log.warn "socket is nil -- failed to send: #{packet.assemble}"
+        err_msg = "Unable to create socket with #{papertrail_addr}"
+        log.error err_msg
+        raise SocketFailureError, err_msg
       else
         begin
           # send it
           @socket.puts packet.assemble
         rescue => e
-          log.error "connection error for #{@papertrail_host}:#{@papertrail_port}: #{e}"
+          err_msg = "Error writing to #{papertrail_addr}: #{e}"
```

Comment:
wondering if this should be warn instead - its not really an error since we will be retrying. 
It would become an error when/if the max retry is reached. 
We can revisit later if this confuses users - looks good now :)

Reply:
good point, when I was digging in the code I found an @error_history.size variable that could help us calculate if we're approaching max retry. I haven't tested that yet but took notes on it in case we want to pursue changes to this again later.